content/en/docs/tasks/manage-kubernetes-objects/kustomization.md

```diff
@@ -86,12 +86,20 @@ metadata:
   name: example-configmap-1-8mbdf7882g
 ```
 
-To generate a ConfigMap from an env file, add an entry to the `envs` list in `configMapGenerator`. Here is an example of generating a ConfigMap with a data item from a `.env` file:
+To generate a ConfigMap from an env file, add an entry to the `envs` list in `configMapGenerator`. This can also be used to set values from local environment variables by omitting the `=` and the value.
+
+{{< note >}}
+It's recommended to use the local environment variable population functionality sparingly - an overlay with a patch is often more maintainable. Setting values from the environment may be useful when they cannot easily be predicted, such as a git SHA.
+{{< /note >}}
+
+Here is an example of generating a ConfigMap with a data item from a `.env` file:
 
 ```shell
 # Create a .env file
+# BAZ will be populated from the local environment variable $BAZ
 cat <<EOF >.env
 FOO=Bar
+BAZ
 EOF
```

Comment:
can you make these names more descriptive? I'd like to avoid "foo", "bar", and "baz". Overall I think this change is worthwhile :)

Reply:
Do you have a concrete suggestion for better names? I was following the convention before me - I would need to update the names in the other sections which I am happy to do if you have ideas, but probably better in a separate PR

Reply:
@geoffcline I just checked and there are 10 instances of `FOO` in the file. The editor would need to carefully check the output matches the new inputs. I feel this is out of scope of this PR.

Reply:
I've thought about this. Unfortunately, I don't feel comfortable approving something that includes lines explicitly prohibited by the style guide https://kubernetes.io/docs/contribute/style/style-guide/#inline-code-formatting

You can ask in slack? or just try to remove "foo" in the examples you are updating?

thanks, Geoffrey

Reply:
Hi @geoffcline,

Firstly, thanks for taking the time to review.

I would strongly argue that the use of `foo` and `bar` is entirely appropriate here. They are explicitly the standard names for something that is "not meaningful and lack[s] context" because any meaning and context would detract from the example. The analogous example would be using `lorem ipsum` in a template example. Anything else is either a) a less common substitute for `lorem ipsum` or b) misleading.

So I see a few ways forward to fixing this missing documentation:
1) suggest better variable names and I will implement them
2) PR the style guide for an exception
  a) I don't love this, because the style guide should be simple and cover the common case - outside of tests and examples of arbitrary variable handling there are not many cases where these would be appropriate names
3) accept this PR to fix the [open issue](https://github.com/kubernetes-sigs/cli-experimental/issues/66) and open a separate issue to build consensus on 1) or 2)
4) move to another channel or tag another reviewer / approver to build consensus
5) accept my argument and "lgtm" :smile: 

TBH I find this pretty hard, I'm contributing docs for something that annoys and even blocks a lot of people in my free time and IMO it feels like an overall improvement being blocked on a nit with no clear suggestion on how to resolve the nit. The contribution matches the existing code style. I can't just remove the `FOO` - the example becomes meaningless.

Thanks again for your time and assistance to resolve
-Aalstair

Reply:
resolving due to no path to resolution and inactivity + separate /lgtm from @jimangel